src/traversals/bfs.jl

```diff
@@ -246,20 +205,33 @@ function examine_neighbor!(visitor::BipartiteVisitor, u::Int, v::Int, vcolor::In
     return visitor.is_bipartite
 end
 
-"""Will return `true` if graph `g` is
-[bipartite](https://en.wikipedia.org/wiki/Bipartite_graph).
 """
-is_bipartite(g::SimpleGraph, s::Int) = _bipartite_visitor(g, s).is_bipartite
+    is_bipartite(g)
+    is_bipartite(g, v)
 
+Will return `true` if graph `g` is [bipartite](https://en.wikipedia.org/wiki/Bipartite_graph).
+If a node `v` is specified, only the connected component to which it belongs is considered.
+"""
 function is_bipartite(g::SimpleGraph)
     cc = filter(x->length(x)>2, connected_components(g))
-    return all(x->is_bipartite(g,x[1]), cc)
+    vmap = Dict{Int,Int}()
+    for c in cc
+        _is_bipartite(g,c[1], vmap=vmap) || return false
+    end
+    return true
 end
 
-function _bipartite_visitor(g::SimpleGraph, s::Int)
+is_bipartite(g::SimpleGraph, v::Int) = _is_bipartite(g, v)
+
+_is_bipartite(g::SimpleGraph, v::Int; vmap = Dict{Int,Int}()) = _bipartite_visitor(g, v, vmap=vmap).is_bipartite
+
+function _bipartite_visitor(g::SimpleGraph, s::Int; vmap=Dict{Int,Int}())
     nvg = nv(g)
     visitor = BipartiteVisitor(nvg)
-    traverse_graph(g, BreadthFirst(), s, visitor)
+    for v in keys(vmap) #have to reset vmap, otherway problems with digraphs
+        vmap[v] = -1
+    end
```

Comment:
each vertex belongs to only one **strongly** connected component, but the problem here is that in digraphs during traversal you can jump from one strongly connected component to another. Maybe something smarter can be done in those cases but at the moment I don't have the time to look deeper into this